### PR TITLE
[WIP] Test suite improvements

### PR DESCRIPTION
**Pull request by D4N**
I have (hopefully) improved some parts of the test suite that were inconvenient before.

1. The variables defined in the config file are now accessible inside the `system_tests` namespace.
2. A new function `p` was added, that converts unix paths to Windows paths on Windows.

### CODE REVIEW

File: tests/bugfixes/github/test_issue_283.py [hunk 1]
```diff
@@ -16,16 +16,15 @@ def check_no_ASAN_UBSAN_errors(self, i, command, got_stderr, expected_stderr):
         Here we want to also check that the two last lines of got_stderr have the expected_stderr
         """
 
-        system_tests.check_no_ASAN_UBSAN_errors(self, i, command, got_stderr, expected_stderr)
+        check_no_ASAN_UBSAN_errors(self, i, command, got_stderr, expected_stderr)
         self.assertListEqual(expected_stderr.splitlines(), got_stderr.splitlines()[-2:])
 
-    filename = os.path.join("$data_path", "pocIssue283.jpg")
+    filename = p("$data_path/pocIssue283.jpg")
```
Comment (piponazo) [inline tests/bugfixes/github/test_issue_283.py]: Please change the name of the function `p` to something more meaningful like: platformPath, uniformPath, convertPath, etc. Anything you come up will be better than simply `p` :wink:

File: tests/system_tests.py [hunk 2]
```diff
@@ -468,6 +474,24 @@ def setUp_file_action(self, expanded_file_name):
         return expanded_file_name
 
 
+def p(path):
```
Comment (piponazo) [inline tests/system_tests.py]: I am not very convinced by this helper. The advantage of using `os.path.join` directly in the test cases is that the developer do not need to think about which slashes he should use, he just would type:

```
filename = os.path.join("$data_path", "pocIssue283.jpg") #Note the lack of slashes here

# To reduce the verbosity we could do
from os.path import join
filename = join("$data_path", "pocIssue283.jpg")
```

Every python developer with some minimum experience will know what `os.path.join` is used for. 

As I commented before, to use a function name like `p` is a bad idea in any programming language. We should always have meaningful names for the functions to help to the code readability.

File: tests/bugfixes/github/test_issue_283.py (hunk 1 above)
Comment (D4N) [reply tests/bugfixes/github/test_issue_283.py]: I wanted to keep it short and concise to not hamper readability, but I agree, `p` is not the best name. What do you think about simply calling it `path`? That would not be awfully long and would further explain that the line is a path.

File: tests/system_tests.py (hunk 2 above)
Comment (D4N) [reply tests/system_tests.py]: My problem with `os.path.join` is that my original intention for the test suite was to keep the required Python knowledge at a minimum and to retain as much of the "look and feel" of the bash scripts. Using `os.path.join` is of course the most Pythonic way of solving this problem, but that will imho only make it more complicated for developers who don't know Python.

Also, I like to keep the slashes inside the string, as the string then is obviously a path.

Comment (piponazo) [reply tests/system_tests.py]: Fair enough. Let's follow the original design idea. If we can keep working like that, I'm happy with the solution.

File: tests/bugfixes/github/test_issue_283.py (hunk 1 above)
Comment (piponazo) [reply tests/bugfixes/github/test_issue_283.py]: That also works for me ;). I'll approve once you change it.

**Comment by D4N** [comment]
I have fixed the name of the function `p` and updated the documentation accordingly. Also, I have included the hooks that I promised on slack and used them as a replacement to `$cat`.

Unfortunately the Appveyor build currently fails at the conan install step:
```
conan install .. --build missing
Auto detecting your dev setup to initialize the default profile (C:\Users\appveyor\.conan\profiles\default)
Found Visual Studio 9
Found Visual Studio 10
Found Visual Studio 11
Found Visual Studio 12
Found Visual Studio 14
Found clang 6.0
Default settings
	os=Windows
	os_build=Windows
	arch=x86_64
	arch_build=x86_64
	compiler=Visual Studio
	compiler.version=14
	build_type=Release
*** You can change them in C:\Users\appveyor\.conan\profiles\default ***
*** Or override with -s compiler='other' -s ...s***
Expat/2.2.5@pix4d/stable: Not found in local cache, looking in remotes...
Expat/2.2.5@pix4d/stable: Trying with 'conan-center'...
ERROR: No JSON object could be decoded
Command exited with code 1
```

**Comment by piponazo** [comment]
> Expat/2.2.5@pix4d/stable: Not found in local cache, looking in remotes...
Expat/2.2.5@pix4d/stable: Trying with 'conan-center'...
ERROR: No JSON object could be decoded

Humm ... It looks like a problem with the bintray servers. Maybe it was a temporary issue. I think I can retrigger the job from appveyor

**Review by piponazo — APPROVED** [review]

**Comment by D4N** [comment]
I'll rename the commit prefixed with experiment, that should retrigger appveyor too.

**Pull request merged**